basket-analysis/basket-analysis-quickstart/Basket_Analysis.ipynb

```diff
@@ -0,0 +1,689 @@
+    "    'outputTable':          'tpch_example.gds.part_similar_to_part'\n",
+    "    }\n",
+    "  ]\n",
+    "});"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e87b0916-9cdd-4377-ae8f-a07c838073e8",
+   "metadata": {
+    "collapsed": false,
+    "name": "cell35"
+   },
+   "source": [
+    "After writing the table, we need to ensure that our current role is allowed to read it.\n",
+    "Alternatively, we can also grant access to all future tables created by the application."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9eb24431-c49e-47b8-bbfa-62dfa46c1be1",
+   "metadata": {
+    "language": "sql",
+    "name": "cell36"
+   },
+   "outputs": [],
+   "source": [
+    "GRANT SELECT ON TPCH_EXAMPLE.GDS.PART_SIMILAR_TO_PART TO ROLE gds_user_role;"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e5b6d40b-67c2-4879-bffe-0c3bcdca5195",
+   "metadata": {
+    "collapsed": false,
+    "name": "cell37"
+   },
+   "source": [
+    "Since the results are now stored in Snowflake, we can query them and join them with our original data.\n",
+    "For example, we can find the names of the most similar parts based on the similarity score.\n",
+    "Simply speaking, this could be used as a recommendation system for parts."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e9f2ca75-2d8b-4971-9bb5-a3be5c034555",
+   "metadata": {
+    "language": "sql",
+    "name": "cell38"
+   },
+   "outputs": [],
+   "source": [
+    "SELECT DISTINCT p_source.p_name, p_target.p_name, sim.similarity\n",
+    "FROM TPCH_EXAMPLE.GDS.PARTS p_source\n",
+    "    JOIN TPCH_EXAMPLE.GDS.PART_SIMILAR_TO_PART sim\n",
+    "        ON p_source.p_partkey = sim.sourcenodeid\n",
+    "    JOIN TPCH_EXAMPLE.GDS.PARTS p_target\n",
+    "        ON p_target.p_partkey = sim.targetnodeid\n",
+    "ORDER BY sim.similarity DESC LIMIT 10;"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1ea473f6-1b3a-4f52-b144-743bcfbdf2bb",
+   "metadata": {
+    "language": "sql",
+    "name": "cell39"
+   },
+   "outputs": [],
+   "source": [
+    "USE ROLE ACCOUNTADMIN;\n",
+    "GRANT OWNERSHIP ON TABLE TPCH_EXAMPLE.GDS.part_similar_to_part TO ROLE gds_user_role REVOKE CURRENT GRANTS;\n",
+    "USE ROLE gds_user_role;"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2043e223-11b6-40f3-998f-1172fd33b516",
+   "metadata": {
+    "collapsed": false,
+    "name": "cell40"
+   },
+   "source": [
+    "The GDS service is a long-running service and should be stopped when not in use.\n",
+    "Once we completed our analysis, we can stop the session, which suspends the container service.\n",
+    "We can restart the session at any time to continue our analysis."
+   ]
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "Streamlit Notebook",
+   "name": "streamlit"
+  },
+  "lastEditStatus": {
+   "authorEmail": "david.fauth@neo4j.com",
```

Comment:
Personal data is spoiled.
The notebook have to be cleaned up before commit, e.g. you can use the following command:
```
jq --indent 1 \
    '
    (.cells[] | select(has("outputs")) | .outputs) = []
    | (.cells[] | select(has("execution_count")) | .execution_count) = null
    | .metadata = {"language_info": {"name":"python", "pygments_lexer": "ipython3"}}
    | .cells[].metadata = {}
    ' path_to_file.ipynb > cleaned.ipynb
```

Reply:
Done.